modules/darwinosl/darwinosl-source.m

```diff
@@ -60,12 +62,73 @@ void darwinosl_sd_options_defaults(DarwinOSLogSourceOptions *self,
            and signs nicely those variables need different kind of care because of ARC */
   __strong OSLogSource *osLogSource;
 } DarwinOSLogSourceDriver;
 
 
+static void
+_log_reader_insert_msg_length_stats(DarwinOSLogSourceDriver *self, gsize len)
+{
+  stats_aggregator_insert_data(self->max_message_size, len);
+  stats_aggregator_insert_data(self->average_messages_size, len);
+}
+
+static void
+_register_aggregated_stats(DarwinOSLogSourceDriver *self)
+{
+  StatsClusterKey sc_key_eps_input;
+  stats_cluster_single_key_legacy_set_with_name(&sc_key_eps_input,
+                                                self->options.super_source_options->super.stats_source | SCS_SOURCE,
+                                                self->super.worker->super.stats_id,
+                                                self->super.worker->super.stats_instance, "processed");
+
+  stats_aggregator_lock();
+  StatsClusterKey sc_key;
+
+  stats_cluster_single_key_legacy_set_with_name(&sc_key,
+                                                self->options.super_source_options->super.stats_source | SCS_SOURCE,
+                                                self->super.worker->super.stats_id,
+                                                self->super.worker->super.stats_instance, "msg_size_max");
+  stats_register_aggregator_maximum(self->options.super_source_options->super.stats_level, &sc_key,
+                                    &self->max_message_size);
+
+  stats_cluster_single_key_legacy_set_with_name(&sc_key,
+                                                self->options.super_source_options->super.stats_source | SCS_SOURCE,
+                                                self->super.worker->super.stats_id,
+                                                self->super.worker->super.stats_instance, "msg_size_avg");
+  stats_register_aggregator_average(self->options.super_source_options->super.stats_level, &sc_key,
+                                    &self->average_messages_size);
+
+  stats_cluster_single_key_legacy_set_with_name(&sc_key,
+                                                self->options.super_source_options->super.stats_source | SCS_SOURCE,
+                                                self->super.worker->super.stats_id,
+                                                self->super.worker->super.stats_instance, "eps");
+
+  stats_register_aggregator_cps(self->options.super_source_options->super.stats_level, &sc_key, &sc_key_eps_input,
+                                SC_TYPE_SINGLE_VALUE,
+                                &self->CPS);
+
+  stats_aggregator_unlock();
+}
+
+static void
+_unregister_aggregated_stats(DarwinOSLogSourceDriver *self)
+{
+  stats_aggregator_lock();
+
+  stats_unregister_aggregator_maximum(&self->max_message_size);
+  stats_unregister_aggregator_average(&self->average_messages_size);
+  stats_unregister_aggregator_cps(&self->CPS);
+
```

Comment:
I hate that we need to add this here. Can't we move these stats registrations to the LogThreadedSourceDriver level?